spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -685,34 +696,74 @@ def save_figure_as(self, fig, fmt):
             save_figure_tofile(fig, fmt, fname)
 
     # ---- Thumbails Handlers
-    def add_thumbnail(self, fig, fmt):
-        thumbnail = FigureThumbnail(background_color=self.background_color)
-        thumbnail.canvas.load_figure(fig, fmt)
+    def _calculate_figure_canvas_width(self, thumbnail):
+        """
+        Calculate the witdh the thumbnail's figure canvas need to have for the
+        thumbnail to fit the scrollarea.
+        """
+        figure_canvas_width = (
+            self.scrollarea.width() -
+            2 * self.lineWidth() -
+            self.scrollarea.viewportMargins().left() -
+            self.scrollarea.viewportMargins().right() -
+            thumbnail.savefig_btn.width() -
+            thumbnail.layout().spacing()
+            )
+        if is_dark_interface():
+            # This is required to take into account some hard-coded padding
+            # and margin in qdarkstyle.
+            figure_canvas_width = figure_canvas_width - 6
+        return figure_canvas_width
 
-        # Scale the thumbnail size, while respecting the figure
-        # dimension ratio.
+    def _setup_thumbnail_size(self, thumbnail):
+        """
+        Scale the thumbnail's canvas size, while respecting its associated
+        figure dimension ratio.
+        """
         fwidth = thumbnail.canvas.fwidth
         fheight = thumbnail.canvas.fheight
-
-        max_length = 100
-        if fwidth/fheight > 1:
-            canvas_width = max_length
+        max_canvas_size = self._calculate_figure_canvas_width(thumbnail)
+        if fwidth / fheight > 1:
+            canvas_width = max_canvas_size
             canvas_height = canvas_width / fwidth * fheight
         else:
-            canvas_height = max_length
+            canvas_height = max_canvas_size
             canvas_width = canvas_height / fheight * fwidth
         thumbnail.canvas.setFixedSize(int(canvas_width), int(canvas_height))
+        thumbnail.layout().setColumnMinimumWidth(0, max_canvas_size)
+
+    def _update_thumbnail_size(self):
+        """
+        Update the thumbnails size so that their width fit that of
+        the scrollarea.
+        """
+        # NOTE: we hide the view widget while we update the thumbnails size
```

Comment:
When is this happening  @jnsebgosselin, I do not follow

Reply:
This happens every time a `Resize` event occur on the scrollarea of the thumbnail scrollbar widget.

Reply:
@goanpeca Is this answering your question or you would like to have more explanation about the hide/show calls enclosing the updating of the thumbnails size?

Reply:
What I mean is that I do not see what is hidden in the animation you posted. Or maybe I am still not understanding 🙃 ?

Reply:
Ok I understand. In reality, nothing is really hidden, it is just a trick to make the resizing animation of the thumbnails look smoother.

The goal here is to prevent a repaint of each thumbnail as soon as their size is updated in the loop, which causes some flickering when the scrollbar is resized. So what I do to prevent that is to hide temporarily all the thumbnails while their size is updated. Once the size of all the thumbnails has been updated, I then show them back so that they are repainted all at once instead of one after the other.

Reply:
Ahhh ok got it. Maybe adding that as a comment makes life easy :-) for next devs